lib/t/feeds.ex

```diff
@@ -633,8 +633,181 @@ defmodule T.Feeds do
     |> Repo.delete_all()
   end
 
+  # Since Live schedule
+
+  def live_schedule do
+    [
+      {_mon = 1, :newbie, [newbies_live_start_at(), newbies_live_end_at()]},
+      {_tue = 2, :newbie, [newbies_live_start_at(), newbies_live_end_at()]},
+      {_wed = 3, :newbie, [newbies_live_start_at(), newbies_live_end_at()]},
+      {_thu = 4, :real, [_start = ~T[19:00:00], _end = ~T[21:00:00]]},
+      {_fri = 5, :newbie, [newbies_live_start_at(), newbies_live_end_at()]},
+      {_sat = 6, :real, [_start = ~T[18:00:00], _end = ~T[20:00:00]]},
+      {_sun = 7, :newbie, [newbies_live_start_at(), newbies_live_end_at()]}
+    ]
+  end
+
+  @doc """
+  Finds out when the next real Since Live will happen:
+
+      iex> mon_12_00 = DateTime.new!(~D[2021-12-20], ~T[12:00:00], "Europe/Moscow")
+      iex> live_next_real_at(_now = mon_12_00)
+      _thu_19_00 = DateTime.new!(~D[2021-12-23], ~T[19:00:00], "Europe/Moscow")
+
+      iex> thu_12_00 = DateTime.new!(~D[2021-12-23], ~T[12:00:00], "Europe/Moscow")
+      iex> live_next_real_at(thu_12_00)
+      _thu_19_00 = DateTime.new!(~D[2021-12-23], ~T[19:00:00], "Europe/Moscow")
+
+      # not very plausible, but asking for the next event the instant
+      # today's event starts, today's event is returned
+      iex> thu_19_00 = DateTime.new!(~D[2021-12-23], ~T[19:00:00], "Europe/Moscow")
+      iex> live_next_real_at(thu_19_00)
+      _thu_19_00 = DateTime.new!(~D[2021-12-23], ~T[19:00:00], "Europe/Moscow")
+
+      # if today's event has started, next event is returned
+      iex> thu_19_00_01 = DateTime.new!(~D[2021-12-23], ~T[19:00:01], "Europe/Moscow")
+      iex> live_next_real_at(thu_19_00_01)
+      _sat_18_00 = DateTime.new!(~D[2021-12-25], ~T[18:00:00], "Europe/Moscow")
+
+      # schedule wraps over to the next week
+      iex> sun_12_00 = DateTime.new!(~D[2021-12-26], ~T[12:00:00], "Europe/Moscow")
+      iex> live_next_real_at(sun_12_00)
+      _thu_19_00 = DateTime.new!(~D[2021-12-30], ~T[19:00:00], "Europe/Moscow")
+
+  """
+  def live_next_real_at(reference \\ DateTime.utc_now()) do
+    weekday_today = Date.day_of_week(reference)
+
+    live_schedule()
+    |> Enum.filter(fn {_weekday, type, _times} -> type == :real end)
+    |> Enum.map(fn {weekday, _type, [start, _end]} ->
+      diff = weekday - weekday_today
+      diff = if diff < 0, do: 7 + diff, else: diff
+
+      reference
+      |> Date.add(diff)
+      |> DateTime.new!(start, "Europe/Moscow")
+    end)
+    |> Enum.sort({:asc, Date})
+    |> Enum.find(fn datetime ->
+      DateTime.compare(datetime, reference) in [:gt, :eq]
+    end)
+  end
```

Comment:
This function can replace `since_live_date`

Reply:
@akhrail1996 this function returns the datetime for the next real live event. I wonder what should be return when it's called during an ongoing live event? Like here:

```elixir
# if today's event has started, next event is returned
iex> thu_19_00_01 = DateTime.new!(~D[2021-12-23], ~T[19:00:01], "Europe/Moscow")
iex> live_next_real_at(thu_19_00_01)
_sat_18_00 = DateTime.new!(~D[2021-12-25], ~T[18:00:00], "Europe/Moscow")
```

Maybe if the event has started but hasn't ended, current event start datetime is returned, but if it has ended, next event datetime is returned?

Reply:
> Maybe if the event has started but hasn't ended, current event start datetime is returned, but if it has ended, next event datetime is returned?

yep